ulog.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "endpoint.h"
+#include "timeout.h"
+
+#define TARGET_SYSTEM_ID 1
+#define SYSTEM_ID 2
+
+#define BUFFER_LEN 2048
+
+class ULog : public Endpoint {
+public:
+    ULog(const char *logs_dir);
+
+    bool start();
+    bool logging_start_timeout();
```

Comment:
Because it will be called from the timeout callback.

```
static bool _ulog_timeout_cb(void *data)
{
    ULog *ulog = static_cast<ULog *>(data);
    return ulog->logging_start_timeout();
}
```